blog-posts/takeuntil-takewhile/takeWhile-takeUntil-takeWhat.md

```diff
@@ -0,0 +1,106 @@
+
+## The Problem
+
+So all in all both code snippets look similar and behave similar, right? Well, No!
+Let's have a look at the marble diagrams describing those operators because this will highlight the difference between those two operators.
+
+![takeUntil marble diagram, kindly provided by Michael Hladky](./assets/takeUntil.png)
+_takeUntil marble diagram, kindly provided by [Michael Hladky](https://twitter.com/Michael_Hladky)_
+
+![takeWhile marble diagram, kindly provided by Michael Hladky](./assets/takeWhile.png)
+_takeWhile marble diagram, kindly provided by [Michael Hladky](https://twitter.com/Michael_Hladky)_
+
+The problem here is that `takeWhile` is intended to take an incoming notification, use some comparison on it, which might lead to an unsubscribe. The important fact is, that `takeWhile` is triggered by the incoming notification and might unsubscribe afterwards. `TakeUntil` in contrast is triggered by the passed Observable.
+That's why `takeWhile` might cause several issues. So definitely, it needs a new notification to unsubscribe. Imagine having a long-living Observable. You will need one notification more with `takeWhile` than with `takeUntil`. Also, this additional notification can initiate multiple processes within your observable. Imagine having code like this:
+
+```ts
+longLivingObservable$
+  .pipe(
+    tap(() => this.startAnimation()),
+    switchMap(val => this.makeHttpCall(val)),
+    takeWhile(val => this.alive),
+  )
+  .subscribe();
+```
+
+So what's the issue with this piece of code? Well, our component is already destroyed and due to the needed notification, needed before unsubscribing kicks in, we will start an animation and trigger an HTTP call. This is probably unwanted and just afterwards we will check if we want to unsubscribe from our Observable. Besides the fact that those operations are totally superfluous, it also might break our app or pollute our state.
+
+_Now maybe one might argue: "Well, I could move the `takeWhile` operator at the very beginning of the observable pipeline!"_
+
+That's true, you could do this, and you will save the unneeded operations, which is a good start, but you won't unsubscribe from inner Observables. So if the Observable returned by `makeHttpCall` is a long-living Observable, it will not unsubscribe from that, if the `takeWhile` is before `switchMap` in the pipe. By the way, the same goes for `takeUntil`, so make sure to have the unsubscribe-operator at the very end of your pipe.
+
+## The Solution
+
+Don't get me wrong, `takeWhile` is an amazing operator, but just if you actually use the incoming value to determine, whether you want to unsubscribe or not! Do not depend on "global" state, when using `takeWhile`.
```

Comment:
Would it make sense to add an real world example using `takeWhile` to strengthen your point and show the difference in usage between `takeUntil` and `takeWhile`?